celavi/scenario.py

```diff
@@ -0,0 +1,569 @@
+"""
+Created January 27, 2022.
+
+@author: rhanes
+"""
+import os
+import sys
+import time
+import yaml
+import pickle
+
+import numpy as np
+import pandas as pd
+
+from scipy.stats import weibull_min
+
+from celavi.routing import Router
+from celavi.costgraph import CostGraph
+from celavi.compute_locations import ComputeLocations
+from celavi.data_filtering import filter_locations, filter_routes
+from celavi.pylca_celavi.des_interface import PylcaCelavi
+from celavi.reeds_importer import ReedsImporter
+from celavi.des import Context
+from celavi.diagnostic_viz import DiagnosticViz
+
+
+class Scenario:
+    """
+    Set up, validate, and execute a CELAVI scenario.
+
+    Execute multiple runs automatically to quantify uncertainty in results.
+    """
+
+    def __init__(self, parser):
+        """
+        Initialize Scenario and validate inputs.
+
+        Parameter
+        ---------
+        parser
+            Parser object for reading in config files.
+

```

Comment:
Consider putting a note about the exceptions this method raises, formatted like this:

```
Raises
-------
IOError
    Raises IOError when...
```